lib/foreman_providers/compute_resource.rb

```diff
@@ -0,0 +1,35 @@
+module ForemanProviders
+  module ComputeResource
+    def self.included(base)
+      base.class_eval do
+        after_create   :create_provider
+        before_destroy :destroy_provider
+
+        private
+
+        def create_provider
+          provider_klass = foreman_type_to_provider_type.constantize
+          provider_klass.create!(
+            :name     => name,
+            :hostname => URI(url).host,
+          ).tap do |ems|
```

Comment:
why use tap? if i read this correctly this will do two sql one to insert and another to update the authentication?

Reply:
:+1: changed it to build up the auth and endpoints then save once